app/src/org/commcare/activities/FormRecordListActivity.java

```diff
@@ -436,16 +443,38 @@ public void run() {
         }
     }
 
-    private void createFormRecordScanResultDialog(Pair<Boolean, String> result) {
+    private void createFormRecordScanResultDialog(Pair<Boolean, String> result, FormRecord record) {
         String title;
         if (result.first) {
             title = Localization.get("app.workflow.forms.scan.title.valid");
         } else {
             title = Localization.get("app.workflow.forms.scan.title.invalid");
         }
         int resId = result.first ? R.drawable.checkmark : R.drawable.redx;
-        showAlertDialog(StandardAlertDialog.getBasicAlertDialogWithIcon(this, title,
-                result.second, resId, null));
+
+        StandardAlertDialog dialog = StandardAlertDialog.getBasicAlertDialogWithIcon(this, title,
+                result.second, resId, null);
+
+        dialog.setNegativeButton(Localization.get("app.workflow.forms.quarantine"), new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                quarantineRecord(record);
+                dismissAlertDialog();
+            }
+        });
+
+        showAlertDialog(dialog);
+    }
+
+    private void quarantineRecord(FormRecord record) {
+        this.formRecordProcessor.updateRecordStatus(record, FormRecord.STATUS_QUARANTINED);
```

Comment:
This is a suggestion-not-requirement comment, but would it be useful to record in the same place as we do with the "quarantined from web" records a message that denotes "Quarantined manually by user" so there's no confusion/ambiguity?